opentracing-shim/src/opentracingshim/__init__.py

```diff
@@ -0,0 +1,248 @@
+    ):
+        # pylint: disable=super-init-not-called
+        self._otel_tracer = tracer
+        if scope_manager is not None:
+            self._scope_manager = scope_manager
+        else:
+            self._scope_manager = ScopeManagerWrapper(self)
+
+    def unwrap(self):
+        """Returns the wrapped OpenTelemetry `Tracer` object."""
+
+        return self._otel_tracer
+
+    @property
+    def scope_manager(self):
+        return self._scope_manager
+
+    @property
+    def active_span(self):
+        span = self._otel_tracer.get_current_span()
+        if span is None:
+            return None
+        return SpanWrapper(self, span.get_context(), span)
+        # TODO: Return a saved instance of SpanWrapper instead of constructing
+        # a new object.
+        # https://github.com/open-telemetry/opentelemetry-python/issues/161#issuecomment-534136274
+
+    @contextmanager
+    def start_active_span(
+        self,
+        operation_name,
+        child_of=None,
+        references=None,
+        tags=None,
+        start_time=None,
+        ignore_active_span=False,
+        finish_on_close=True,
+    ) -> ScopeWrapper:
+        span = self.start_span(
+            operation_name=operation_name,
+            child_of=child_of,
+            references=references,
+            tags=tags,
+            start_time=start_time,
+            ignore_active_span=ignore_active_span,
+        )
+        with self._scope_manager.activate(span, finish_on_close) as scope:
+            yield scope
+
+    def start_span(
+        self,
+        operation_name=None,
+        child_of=None,
+        references=None,
+        tags=None,
+        start_time=None,
+        ignore_active_span=False,
+    ) -> SpanWrapper:
+        parent = child_of
+        if parent is not None:
+            if isinstance(parent, SpanWrapper):
+                parent = child_of.unwrap()
+            elif isinstance(parent, SpanContextWrapper):
+                parent = child_of.unwrap()
+            else:
+                raise RuntimeError(
+                    "Invalid parent type when calling start_span()."
+                )
+            # TODO: Should we use the OpenTracing base class for the type check?
+
+        # Use active span as parent when no explicit parent is specified.
+        if (
+            self.active_span is not None
+            and not ignore_active_span
+            and not parent
+        ):
+            parent = self.active_span.unwrap()
+
+        span = self._otel_tracer.create_span(operation_name, parent)
+
+        if references:
+            if not isinstance(references, list):
+                references = [references]
+            for ref in references:
+                span.add_link(ref.referenced_context.unwrap())
+
+        if tags:
+            for key, value in tags.items():
+                span.set_attribute(key, value)
+
+        span.start(start_time=start_time)
```

Comment:
I'm not totally sure but I think both times are in different units:
Opentracing uses epoch as float in seconds: https://github.com/opentracing/opentracing-python/blob/7d2e62b5999d3cdec9a9c2720a34a1d11f425461/opentracing/span.py#L117
Opentelemetry uses epoch as int in nanoseconds https://github.com/open-telemetry/opentelemetry-python/blob/3b37675aa65b024eed15840f5b416d444213e5c2/opentelemetry-sdk/src/opentelemetry/sdk/trace/__init__.py#L252

Reply:
Great catch. I've handled this by converting seconds to nanoseconds and vice versa.